python_modules/dagster/dagster/_core/definitions/asset_spec.py

```diff
@@ -18,6 +19,19 @@
 if TYPE_CHECKING:
     from dagster._core.definitions.asset_dep import AssetDep
 
+# SYSTEM_METADATA_KEY_ASSET_VARIETAL lives on the metadata of an asset
+# (which currently ends up on the Output associated with the asset key)
+# whih encodes the "varietal" of asset. "Unexecutable" assets are assets
+# that cannot be materialized in Dagster, but can have events in the event
+# log keyed off of them, making Dagster usable as a observability and lineage tool
+# for externally materialized assets.
+SYSTEM_METADATA_KEY_ASSET_VARIETAL = "dagster/asset_varietal"
+
+
+class AssetVarietal(Enum):
+    UNEXECUTABLE = "UNEXECUTABLE"
+    MATERIALIZEABLE = "MATERIALIZEABLE"
```

Comment:
nit: varietal -> execution type, fn type, op type, capability 
varietal feels like niche language to use here 

some concern about incorrect eager abstraction, so would err on the side of covering known conditions versus trying to cover notions in expected future changes

Reply:
Mind if we do rename downstack after graphql pr

Reply:
thats fine